common/lib/xmodule/xmodule/video_module/video_module.py

```diff
@@ -734,3 +735,78 @@ def _update_transcript_for_index(language=None):
         xblock_body["content_type"] = "Video"
 
         return xblock_body
+
+    @property
+    def request_cache(self):
+        """
+        Returns the request_cache from the runtime.
+        """
+        return self.runtime.service(self, "request_cache")
+
+    @memoize_in_request_cache('request_cache')
+    def get_cached_val_data_for_course(self, video_profile_names, course_id):
+        """
+        Returns the VAL data for the requested video profiles for the given course.
+        """
+        return edxval_api.get_video_info_for_course_and_profiles(unicode(course_id), video_profile_names)
+
+    def student_view_json(self, context):
+        """
+        Returns a JSON representation of the student_view of this XModule.
+        The contract of the JSON content is between the caller and the particular XModule.
+        """
+        # Honor only_on_web
```

Comment:
It's an "honorable" way of saying:

`If the "only_on_web" field is set on this video, do not return the rest of the video's data in this json view, since the video is meant to be accessed only through its web view."